exporter/awsxrayexporter/internal/translator/segment.go

```diff
@@ -60,16 +61,227 @@ const (
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the name
+	if myAwsRemoteService, ok := span.Attributes().Get(awsRemoteService); ok {
+		dependencySubsegment.Name = awsxray.String(myAwsRemoteService.Str())
+	}
+
+	// Make this a subsegment
+	dependencySubsegment.Type = awsxray.String("subsegment")
+
+	if dependencySubsegment.Namespace == nil {
+		dependencySubsegment.Namespace = awsxray.String("remote")
+	}
+
+	// Remove span links from consumer spans
+	if span.Kind() == ptrace.SpanKindConsumer {
+		dependencySubsegment.Links = nil
+	}
+
+	return dependencySubsegment, err
+}
+
+func MakeServiceSegment(span ptrace.Span, resource pcommon.Resource, indexedAttrs []string, indexAllAttrs bool, logGroupNames []string, skipTimestampValidation bool, serviceSegmentID pcommon.SpanID) (*awsxray.Segment, error) {
+	// We always create a segment for the service
+	var serviceSpan ptrace.Span = ptrace.NewSpan()
+	span.CopyTo(serviceSpan)
+
+	// Set the span id to the one internally generated
+	serviceSpan.SetSpanID(serviceSegmentID)
+
+	for _, v := range removeAnnotationsFromServiceSegment {
+		serviceSpan.Attributes().Remove(v)
+	}
+
+	serviceSegment, err := MakeSegment(serviceSpan, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the name
+	if myAwsLocalService, ok := span.Attributes().Get(awsLocalService); ok {
+		serviceSegment.Name = awsxray.String(myAwsLocalService.Str())
+	}
+
+	// Remove the HTTP field
+	serviceSegment.HTTP = nil
+
+	// Remove AWS subsegment fields
+	serviceSegment.AWS.Operation = nil
+	serviceSegment.AWS.AccountID = nil
+	serviceSegment.AWS.RemoteRegion = nil
+	serviceSegment.AWS.RequestID = nil
+	serviceSegment.AWS.QueueURL = nil
+	serviceSegment.AWS.TableName = nil
+	serviceSegment.AWS.TableNames = nil
+
+	// Delete all metadata that does not start with 'otel.resource.'
+	for _, metaDataEntry := range serviceSegment.Metadata {
+		for key := range metaDataEntry {
+			if !strings.HasPrefix(key, "otel.resource.") {
+				delete(metaDataEntry, key)
+			}
+		}
+	}
+
+	// Make it a segment
+	serviceSegment.Type = nil
+
+	// Remote namespace
+	serviceSegment.Namespace = nil
+
+	// Remove span links from non-consumer spans
+	if span.Kind() != ptrace.SpanKindConsumer {
+		serviceSegment.Links = nil
+	}
+
+	return serviceSegment, nil
+}
+
+func MakeServiceSegmentWithoutDependency(span ptrace.Span, resource pcommon.Resource, indexedAttrs []string, indexAllAttrs bool, logGroupNames []string, skipTimestampValidation bool) ([]*awsxray.Segment, error) {
+	segment, err := MakeSegment(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Make internal spans a segment
+	// And Ensure consumer process spans are not made a segment
+	_, hasAwsRemoteService := span.Attributes().Get(awsRemoteService)
+
+	if hasAwsRemoteService {
+		segment.Type = nil
+	}
```

Comment:
Probably want to clear `namespace`, too.

Reply:
This was an additional requirement, I have added it.